src/checkpoint/serializers/sanitizer.cc

```diff
@@ -0,0 +1,68 @@
+/*
+//@HEADER
+// *****************************************************************************
+//
+//                                 sanitizer.cc
+//                           DARMA Toolkit v. 1.0.0
+//                 DARMA/checkpoint => Serialization Library
+//
+// Copyright 2019 National Technology & Engineering Solutions of Sandia, LLC
+// (NTESS). Under the terms of Contract DE-NA0003525 with NTESS, the U.S.
+// Government retains certain rights in this software.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of source code must retain the above copyright notice,
+//   this list of conditions and the following disclaimer.
+//
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+//
+// * Neither the name of the copyright holder nor the names of its
+//   contributors may be used to endorse or promote products derived from this
+//   software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact darma@sandia.gov
+//
+// *****************************************************************************
+//@HEADER
+*/
+
+#if !defined INCLUDED_CHECKPOINT_SERIALIZERS_SANITIZER_CC
+#define INCLUDED_CHECKPOINT_SERIALIZERS_SANITIZER_CC
+
+#include "checkpoint/serializers/sanitizer.h"
+
+#include <memory>
+
+namespace checkpoint { namespace sanitizer {
+
+Runtime* rt() {
+  static std::unique_ptr<sanitizer::Runtime> base_rt;
+  if (base_rt == nullptr) {
+    base_rt = std::make_unique<sanitizer::Runtime>();
+  }
+  return base_rt.get();
+}
+
+bool enabled() {
+  return false;
+}
```

Comment:
Would it make sense to have this be a member of the `Runtime` object?

Reply:
These have to be C functions because they are LD preloaded by the serializer runtime.

Reply:
Could you point me at where you picked up that constraint? That doesn't sound right to me, since `LD_PRELOAD` should just be importing whatever symbols it finds in a `foo.so` library into the list that `ld.so` will resolve against. If those are C function names or C++ mangled names shouldn't matter.

That said, this isn't a big deal.